<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ColumnDialog</class>
 <widget class="QDialog" name="ColumnDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>424</width>
    <height>360</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Column</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="columnGroup">
     <property name="title">
      <string>Name and type</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="2">
       <widget class="NumericSpinBox" name="scale">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>,</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="NumericSpinBox" name="precision">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="typeLabel">
        <property name="text">
         <string>Data type:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="nameLabel">
        <property name="text">
         <string>Column name:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="name"/>
      </item>
      <item row="0" column="2" colspan="3">
       <widget class="QLabel" name="sizeLabel">
        <property name="text">
         <string>Size:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="typeCombo">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="constraintsGroup">
     <property name="title">
      <string>Constraints</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QStackedWidget" name="constraintModesWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="simplePage">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="3" column="0">
           <widget class="QCheckBox" name="uniqueCheck">
            <property name="text">
             <string>Unique</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QToolButton" name="pkButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="fkCheck">
            <property name="text">
             <string>Foreign Key</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QToolButton" name="fkButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QCheckBox" name="collateCheck">
            <property name="text">
             <string>Collate</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QCheckBox" name="notNullCheck">
            <property name="text">
             <string>Not NULL</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QCheckBox" name="checkCheck">
            <property name="text">
             <string>Check condition</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="pkCheck">
            <property name="text">
             <string>Primary Key</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QCheckBox" name="defaultCheck">
            <property name="text">
             <string>Default</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QToolButton" name="uniqueButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QToolButton" name="checkButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QToolButton" name="notNullButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QToolButton" name="collateButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QToolButton" name="defaultButton">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="advancedPage">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QToolBar" name="constraintsToolbar"/>
          </item>
          <item>
           <widget class="QTableView" name="constraintsView">
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="horizontalScrollMode">
             <enum>QAbstractItemView::ScrollPerPixel</enum>
            </property>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="bottomWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QCheckBox" name="advancedCheck">
        <property name="text">
         <string>Advanced mode</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDialogButtonBox" name="buttonBox">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="standardButtons">
         <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NumericSpinBox</class>
   <extends>QSpinBox</extends>
   <header>common/numericspinbox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>name</tabstop>
  <tabstop>typeCombo</tabstop>
  <tabstop>scale</tabstop>
  <tabstop>precision</tabstop>
  <tabstop>pkCheck</tabstop>
  <tabstop>pkButton</tabstop>
  <tabstop>fkCheck</tabstop>
  <tabstop>fkButton</tabstop>
  <tabstop>uniqueCheck</tabstop>
  <tabstop>uniqueButton</tabstop>
  <tabstop>checkCheck</tabstop>
  <tabstop>checkButton</tabstop>
  <tabstop>notNullCheck</tabstop>
  <tabstop>notNullButton</tabstop>
  <tabstop>collateCheck</tabstop>
  <tabstop>collateButton</tabstop>
  <tabstop>defaultCheck</tabstop>
  <tabstop>defaultButton</tabstop>
  <tabstop>advancedCheck</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>constraintsView</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ColumnDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ColumnDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
